PopupBridge/src/main/java/com/braintreepayments/api/PopupBridgeClient.kt

```diff
@@ -46,92 +51,90 @@ class PopupBridgeClient @SuppressLint("SetJavaScriptEnabled") @VisibleForTesting
     }
 
-    private fun runJavaScriptInWebView(script: String) {
-        val webView = webViewRef.get() ?: return
-        webView.post(Runnable {
-            val webViewInternal = webViewRef.get() ?: return@Runnable
-            webViewInternal.evaluateJavascript(script, null)
-        })
+    fun handleReturnToApp(intent: Intent) {
+        val pendingRequest = pendingRequestRepository.getPendingRequest()
+        if (pendingRequest == null) return
+
+        pendingRequestRepository.clearPendingRequest()
+
+        when (val browserSwitchFinalResult = browserSwitchClient.completeRequest(intent, pendingRequest)) {
+            is BrowserSwitchFinalResult.Success -> runSuccessJavaScript(browserSwitchFinalResult.returnUrl)
+            is BrowserSwitchFinalResult.Failure -> runCanceledJavaScript()
+            is BrowserSwitchFinalResult.NoResult -> runCanceledJavaScript()
+        }
     }
 
-    fun onBrowserSwitchResult(result: BrowserSwitchResult) {
+    private fun runSuccessJavaScript(returnUri: Uri) {
+        if (returnUri.host != POPUP_BRIDGE_URL_HOST) return
+
         var error: String? = null
-        var payload: String? = null
-
-        val returnUri = result.deepLinkUrl
-        if (result.status == BrowserSwitchStatus.CANCELED) {
-            runJavaScriptInWebView(
-                (""
-                        + "function notifyCanceled() {"
-                        + "  if (typeof window.popupBridge.onCancel === 'function') {"
-                        + "    window.popupBridge.onCancel();"
-                        + "  } else {"
-                        + "    window.popupBridge.onComplete(null, null);"
-                        + "  }"
-                        + "}"
-                        + ""
-                        + "if (document.readyState === 'complete') {"
-                        + "  notifyCanceled();"
-                        + "} else {"
-                        + "  window.addEventListener('load', function () {"
-                        + "    notifyCanceled();"
-                        + "  });"
-                        + "}")
-            )
-            return
-        } else if (result.status == BrowserSwitchStatus.SUCCESS) {
-            if (returnUri == null || returnUri.host != POPUP_BRIDGE_URL_HOST) {
-                return
-            }
 
-            val json = JSONObject()
-            val queryItems = JSONObject()
-
-            val queryParams = returnUri.queryParameterNames
-            if (queryParams.isNotEmpty()) {
-                for (queryParam in queryParams) {
-                    try {
-                        queryItems.put(queryParam, returnUri.getQueryParameter(queryParam))
-                    } catch (e: JSONException) {
-                        error = "new Error('Failed to parse query items from return URL. " +
-                                e.localizedMessage + "')"
-                    }
-                }
-            }
+        val payLoadJson = JSONObject()
```

Comment:
Super nit but should this be cased as `payloadJson` instead? That's typically how I've seen payload within our codebases.